src/finn/custom_op/fpgadataflow/rtl/thresholding_rtl.py

```diff
@@ -0,0 +1,794 @@
+        depth_trigger_bram = self.get_nodeattr("depth_trigger_bram")
+        depth_trigger_uram = self.get_nodeattr("depth_trigger_uram")
+        pe = self.get_nodeattr("PE")
+        ret = self.get_pe_mem_geometries()
+        for mem_cfg in ret:
+            (width, depth) = mem_cfg
+            primitives = mem_primitives_versal
+            if depth_trigger_bram != 0 or depth_trigger_uram != 0:
+                if depth >= depth_trigger_bram and depth < depth_trigger_uram:
+                    primitives = {k: v for (k, v) in mem_primitives_versal.items() if "BRAM" in k}
+                elif depth >= depth_trigger_uram:
+                    primitives = {k: v for (k, v) in mem_primitives_versal.items() if "URAM" in k}
+            alts = get_memutil_alternatives(mem_cfg, primitives)
+            primary_alt = alts[0]
+            res_type = primary_alt[0].split("_")[0]
+            res_count, eff, waste = primary_alt[1]
+            res_dict[res_type] = res_dict.get(res_type, 0) + pe * res_count
+        return res_dict
+
+    def calc_tmem(self):
+        """Calculates and returns TMEM."""
+        num_channels = self.get_nodeattr("NumChannels")
+        pe = self.get_nodeattr("PE")
+        return num_channels // pe
+
+    def infer_node_datatype(self, model):
+        """Used for FINN DataType inference: set the output tensors' datatypes
+        accordingly for this node"""
+        node = self.onnx_node
+        idt = model.get_tensor_datatype(node.input[0])
+        if idt != self.get_input_datatype():
+            warn_str = "inputDataType changing for %s: %s -> %s " % (
+                node.name,
+                str(self.get_input_datatype().name),
+                str(idt.name),
+            )
+            warnings.warn(warn_str)
+        self.set_nodeattr("inputDataType", idt.name)
+        # set output datatype from property
+        odt = self.get_output_datatype()
+        model.set_tensor_datatype(node.output[0], odt)
+
+    def verify_node(self):
+        """Required by the FINN nalysis module. Checks if custom ops in graph
+        are correctly built, with all attributes and inputs."""
+        return []
+
+    def bram_estimation(self):
+        res_dict = self.get_memory_estimate()
+        return res_dict.get("BRAM", 0)
+
+    def uram_estimation(self):
+        res_dict = self.get_memory_estimate()
+        return res_dict.get("URAM", 0)
+
+    def lut_estimation(self):
+        res_dict = self.get_memory_estimate()
+        return res_dict.get("LUTRAM", 0)
+
+    def get_input_datatype(self, ind=0):
+        return DataType[self.get_nodeattr("inputDataType")]
+
+    def get_output_datatype(self, ind=0):
+        return DataType[self.get_nodeattr("outputDataType")]
+
+    def get_weight_datatype(self):
+        """The term 'weights' and 'thresholds' are used interchangably in this class."""
+        return DataType[self.get_nodeattr("weightDataType")]
+
+    def minimize_accumulator_width(self, model):
```

Comment:
This also looks really close to `minimize_accumulator_width` that is defined in `thresholding.py` and it shouldn't be necessary to redefine it here if the behavior is the same in HLS and RTL variant.